plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,538 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_l3out_interface_group_policy
+short_description: Manage L3Out Interface Group Policy on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage L3Out Interface Group Policy on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be an L3Out template.
+    type: str
+    aliases: [ l3out_template ]
+  template_id:
+    description:
+    - The ID of the L3Out template.
+    type: str
+    aliases: [ l3out_template_id ]
+  l3out:
+    description:
+    - The name of the L3Out.
+    type: str
+    aliases: [ l3out_name ]
+  l3out_uuid:
+    description:
+    - The UUID of the L3Out.
+    type: str
+  description:
+    description:
+    - The description of the L3Out Interface Group Policy.
+    type: str
+  name:
+    description:
+    - The name of the L3Out Interface Group Policy.
+    type: str
+    aliases: [ l3out_interface_group_policy ]
+  interface_routing_policy:
```

Comment:
The `Interface Routing Policy` is unique within the tenant level, so only `Interface Routing Policy` name is enough. Please recheck the logic once again.